src/ccstruct/normalis.cpp

```diff
@@ -163,8 +162,8 @@ static void ComputeRunlengthImage(const TBOX &box,
   // Set a 2-d image array to the run lengths at each pixel.
   for (int ix = 0; ix < width; ++ix) {
     int y = 0;
-    for (int i = 0; i < y_coords[ix].size(); ++i) {
-      int y_edge = ClipToRange(y_coords[ix][i], 0, height);
+    for (auto y_coord : y_coords[ix]) {
```

Comment:
If ClipToRange does not change first arg, copies may be expensive here (use `auto &`).

Reply:
`y_coord` is of type `int`, to this should be fine.